src/geometry_aux.cpp

```diff
@@ -159,7 +159,8 @@ neighbor_lists()
   write_message("Building neighboring cells lists for each surface...", 6);
 
   for (int i = 0; i < cells.size(); i++) {
-    for (auto token : cells[i]->region_) {
+    CSGCell* c= (CSGCell*)cells[i];
+    for (auto token : c->region_) {
```

Comment:
We prefer the new C++ style casts to C-style casts like these.  The proper one to use in this case would be a `dynamic_cast`, and then you should make sure the cast was successful by checking if the `dynamic_cast` gave you back a null pointer.  I don't think the mess is worth it though.  I think we should just leave `region_` and `rpn_` as members of the `Cell` class and make a more thorough cleanup later (outside of this PR).

Reply:
Yeah I think you're right. I'll remove the commits changing this. I'd rather leave those as part of `Cell` than have to do casting at all.